torchvision/csrc/cpu/video/Video.cpp

```diff
@@ -0,0 +1,393 @@
+  }
+  return sizeof(T);
+}
+
+size_t fillVideoTensor(
+    DecoderOutputMessage& msgs,
+    torch::Tensor& videoFrame,
+    torch::Tensor& videoFramePts) {
+  return fillTensorList<uint8_t>(msgs, videoFrame, videoFramePts);
+}
+
+size_t fillAudioTensor(
+    DecoderOutputMessage& msgs,
+    torch::Tensor& audioFrame,
+    torch::Tensor& audioFramePts) {
+  return fillTensorList<float>(msgs, audioFrame, audioFramePts);
+}
+
+std::string parse_type_to_string(const std::string& stream_string) {
+  static const std::array<std::pair<std::string, MediaType>, 4> types = {{
+      {"video", TYPE_VIDEO},
+      {"audio", TYPE_AUDIO},
+      {"subtitle", TYPE_SUBTITLE},
+      {"cc", TYPE_CC},
+  }};
+  auto device = std::find_if(
+      types.begin(),
+      types.end(),
+      [stream_string](const std::pair<std::string, MediaType>& p) {
+        return p.first == stream_string;
+      });
+  if (device != types.end()) {
+    return device->first;
+  }
+  AT_ERROR("Expected one of [audio, video, subtitle, cc] ", stream_string);
+}
+
+MediaType parse_type_to_mt(const std::string& stream_string) {
+  static const std::array<std::pair<std::string, MediaType>, 4> types = {{
+      {"video", TYPE_VIDEO},
+      {"audio", TYPE_AUDIO},
+      {"subtitle", TYPE_SUBTITLE},
+      {"cc", TYPE_CC},
+  }};
+  auto device = std::find_if(
+      types.begin(),
+      types.end(),
+      [stream_string](const std::pair<std::string, MediaType>& p) {
+        return p.first == stream_string;
+      });
+  if (device != types.end()) {
+    return device->second;
+  }
+  AT_ERROR("Expected one of [audio, video, subtitle, cc] ", stream_string);
+}
+
+std::tuple<std::string, long> _parseStream(const std::string& streamString) {
+  TORCH_CHECK(!streamString.empty(), "Stream string must not be empty");
+  static const std::regex regex("([a-zA-Z_]+)(?::([1-9]\\d*|0))?");
+  std::smatch match;
+
+  TORCH_CHECK(
+      std::regex_match(streamString, match, regex),
+      "Invalid stream string: '",
+      streamString,
+      "'");
+
+  std::string type_ = "video";
+  type_ = parse_type_to_string(match[1].str());
+  long index_ = -1;
+  if (match[2].matched) {
+    try {
+      index_ = c10::stoi(match[2].str());
+    } catch (const std::exception&) {
+      AT_ERROR(
+          "Could not parse device index '",
+          match[2].str(),
+          "' in device string '",
+          streamString,
+          "'");
+    }
+  }
+  return std::make_tuple(type_, index_);
+}
+
+void Video::_getDecoderParams(
+    int64_t videoStartS,
+    int64_t getPtsOnly,
+    std::string stream,
+    long stream_id = -1,
+    bool all_streams = false,
+    double seekFrameMarginUs = 10) {
+  int64_t videoStartUs = int64_t(videoStartS * 1e6);
+
+  params.timeoutMs = decoderTimeoutMs;
+  params.startOffset = videoStartUs;
+  params.seekAccuracy = 10;
+  params.headerOnly = false;
+
+  params.preventStaleness = false; // not sure what this is about
+
+  if (all_streams == true) {
```

Comment:
Can you remind me again when this gets enabled?

Reply:
added to discussion points

Reply:
this only happens at init